src/v3/preact.config.js

```diff
@@ -128,76 +121,60 @@ export default {
-      '..',
-      '..',
+    config.resolve.alias['@okta/okta-auth-js'] = rootResolve(
       'node_modules',
       '@okta',
       'okta-auth-js',
       'esm',
       'esm.browser.js',
     );
 
-    // broadcast-channel esnode bundle is loaded by default, browser one should be used
-    config.resolve.alias['broadcast-channel'] = resolve(
-      '..',
-      '..',
+    // broadcast-channel esnode bundle is loaded by default, browser one should
+    // be used
+    config.resolve.alias['broadcast-channel'] = rootResolve(
       'node_modules',
       'broadcast-channel',
       'dist',
       'esbrowser',
       'index.js',
     );
 
-    if (env.production) {
-      config.output.libraryTarget = 'umd';
-    }
-
     // polyfills for jsonforms
-    config.plugins.push(
-      new DefinePlugin({
-        // https://github.com/APIDevTools/json-schema-ref-parser#browser-support
-        process: {
-          browser: true,
-        },
-        Buffer: buffer,
-      }),
-    );
-
-    config.devServer = {
-      host: HOST,
-      static: [PLAYGROUND, TARGET, {
-        staticOptions: {
-          watchContentBase: true
-        }
-      }],
-      historyApiFallback: true,
-      // FIXME CSP prevents scripts from loading in dev mode
-      // headers: { 'Content-Security-Policy': `script-src http://${HOST}:${DEV_SERVER_PORT}` },
-      compress: true,
-      port: DEV_SERVER_PORT,
-      proxy: [{
-        context: [
-          '/oauth2/',
-          '/api/v1/',
-          '/idp/idx/',
-          '/login/getimage',
-          '/sso/idps/',
-          '/app/UserHome',
-          '/oauth2/v1/authorize',
-          '/auth/services/',
-          '/.well-known/webfinger'
-        ],
-        target: `http://${HOST}:${MOCK_SERVER_PORT}`
-      }],
-    },
+    // https://github.com/APIDevTools/json-schema-ref-parser#browser-support
+    config.plugins.push(new DefinePlugin({ process: { browser: true }, Buffer }));
 
     // preact-cli-plugin-env-vars
     envVars(config, env, helpers);
-    /* eslint-enable no-param-reassign */
+
+    // override with
+    Object.assign(config, webpackMerge(config, omit(playgroundConfig, [
+      'devServer.headers.Content-Security-Policy', // merge instead of override
+      'devServer.port',
+      'entry',
+      'module',
+      'output',
+      'resolve.extensions',
+    ]), {
+      // TODO use built assets instead of relying on dev server
+      // add entry for playground bundle
+      entry: {
+        playground: rootResolve('playground', 'main.ts'),
+      },
+      devtool: 'cheap-module-source-map',
+      devServer: {
+        headers: {
+          'Content-Security-Policy': mergeContentSecurityPolicies(
+            "object-src 'self'; script-src 'self' 'unsafe-eval'",
+            `script-src ${config.devServer.host}:${config.devServer.port}`,
+            playgroundConfig.devServer.headers['Content-Security-Policy'] || '',
+          ),
+        },
+      },
+    }));
```

Comment:
There's a bunch of nesting here that makes it hard to parse what is happening, could it be broken up into smaller statements?